pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1605,40 +1691,74 @@ public CompletableFuture<Void> close(
             }
         }
 
-        CompletableFuture<Void> clientCloseFuture = closeWithoutWaitingClientDisconnect
-                ? CompletableFuture.completedFuture(null)
-                : FutureUtil.waitForAll(futures);
+        CompletableFuture<Void> disconnectClientsInCurrentCall = null;
+        // Note: "disconnectClientsToCache" is a non-able value, it is null when close type is transferring.
+        MutableObject<CompletableFuture<Void>> disconnectClientsToCache = new MutableObject<>();
```

Comment:
It's better to use AtomicReference instead of MutableObject since the value is passed to another thread.

Reply:
Done, and added another change [here](https://github.com/apache/pulsar/pull/17524/commits/996cebaf2a156d6881c00e5fa1574c42f45a30a7#diff-5edf14cc6f25857d0cfdd26b2d3b3141230ecfb0dfa95aebf7583fd76ede4c4bR1752-R1757)

Reply:
LGTM